### PR TITLE
complete flattenDDF with bugs fixed for corner cases plus more comprehensive tests

### PR DESCRIPTION
**Pull request by namma**
- Correct column name to have valid DDF from JSON
- Fix Bugs
- Add more tests using sleep_data and Smurf event logs. 
- Refactor code

### CODE REVIEW

**Comment by khangich** [comment]
ok to test

File: spark/src/test/scala/io/spark/ddf/content/NestedDDFSuite.scala [hunk 1]
```diff
@@ -46,51 +68,78 @@ class NestedDDFSuite extends ATestSuite {
...
 
-    q = s"""
-    select _id.oid as id.oid,cid,created_at.date,data.bookmarkTime,data.isFirstSleepOfDay,
-      data.normalizedSleepQuality,data.realDeepSleepTimeInMinutes,data.realEndTime,data.realSleepTimeInMinutes,
-      data.realStartTime,data.sleepStateChanges,itype,ts,u_at.date,uid.oid
-    from
-    """
-    val ddf: DDF = manager.newDDF(manager, df, Array(classOf[DataFrame]), null, null, SparkUtils.schemaFromDataFrame(df))
-    println("ddf schema: \n" + ddf.getSchema.toString)
+  test("get flatten DDF loaded from Smurf pubsub JSON file") {
```
Comment (binhmop) [inline spark/src/test/scala/io/spark/ddf/content/NestedDDFSuite.scala]: Please remove this asap @namma

Comment (namma) [inline spark/src/test/scala/io/spark/ddf/content/NestedDDFSuite.scala]: sorry, I forgot to remove the code. Data was not committed fortunately.
